internal/service/organization/resource_organization_test.go

```diff
@@ -56,39 +63,71 @@ func testAccCheckMongoDBAtlasOrganizationExists(resourceName string) resource.Te
 			return fmt.Errorf("no ID is set")
 		}
 
+		if rs.Primary.Attributes["public_key"] == "" {
+			return fmt.Errorf("no public_key is set")
+		}
+
+		if rs.Primary.Attributes["private_key"] == "" {
+			return fmt.Errorf("no private_key is set")
+		}
+
 		ids := conversion.DecodeStateID(rs.Primary.ID)
 
-		organizationOptions := &matlas.OrganizationsListOptions{}
-		orgs, _, err := conn.Organizations.List(context.Background(), organizationOptions)
+		cfg := config.Config{
+			PublicKey:  rs.Primary.Attributes["public_key"],
+			PrivateKey: rs.Primary.Attributes["private_key"],
+			BaseURL:    acc.TestAccProviderSdkV2.Meta().(*config.MongoDBClient).Config.BaseURL,
+		}
```

Comment:
side question: how was it working before with previous SDK without using public and private key?

Reply:
oh maybe it's because you need to change creds here? if so maybe worth adding a comment

Reply:
Since the API keys being used to call the ListOrganizations() was not associated with the newly created org, it would not be returned, hence we now use the API keys of the new org - https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/2023-11-15/#tag/Organizations/operation/listOrganizations To use this resource, the requesting API Key must have the Organization Member role
Will add a comment

Reply:
sounds good, feel free to resolve after you add the comment.

Reply:
moved to a new method and added comment, resolving